File: python/11/main.py
def maxArea(height) -> int:
    index1 = ptr1 = 0
    index2 = ptr2 = len(height)-1
    containerSize = min(height[index1], height[index2]) * (index2 - index1)
    while(ptr1 != ptr2):
        if(height[ptr1] < height[ptr2]):
            ptr1 += 1
        else:
            ptr2 -= 1
        newContainerSize = min(height[ptr1], height[ptr2]) * (ptr2 - ptr1)
        if newContainerSize > containerSize:
            index1 = ptr1
            index2 = ptr2
            containerSize = newContainerSize
    return containerSize
print(maxArea([1,1]))

File: typescript/11/main.ts
function maxArea(height: number[]): number {
  let left = 0,
    right = height.length - 1;
  let max = Math.min(height[left], height[right]) * (right - left);
  while (left < right) {
    if (height[left] < height[right]) {
      left++;
    } else {
      right--;
    }
    max = Math.max(max, Math.min(height[left], height[right]) * (right - left));
  }
  return max;
}

console.log(maxArea([1, 8, 6, 2, 5, 4, 8, 3, 7]));
